base/socket.jl

```diff
@@ -282,10 +282,17 @@ mutable struct TCPSocket <: LibuvStream
         return tcp
     end
 end
-function TCPSocket()
+function TCPSocket(; delay=true)  # kw arg "delay": if true, libuv delays creation of the socket
+                                  # fd till the first bind call
+
```

Comment:
could we avoid starting the function with an empty line? it's not the usual formatting style